src/main/java/duke/core/Parser.java

```diff
@@ -0,0 +1,188 @@
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.InvalidCommandException;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.tasks.TaskList;
+
+/**
+ * This class represents a parser that reads user input and converts it into Commands
+ */
+public class Parser {
+
+    /**
+     * Takes in user input and parses it into a Command that can be executed.
+     * TaskList-specific Commands will be executed on the given TaskList.
+     *
+     * @param userInput The String containing the user input.
+     * @param taskList  The TaskList that certain Commands will be executed on.
+     * @return The Command generated by parsing the user input.
+     * @throws InvalidCommandException if the user input is not a valid command.
+     */
+    public static Command parseUserInput(String userInput, TaskList taskList) throws InvalidCommandException {
+
+        //Check first word
+        String[] commandAndRemaining = userInput.strip().split(" ", 2);
+        String commandUpper = commandAndRemaining[0].toUpperCase();
+        Command cmd;
+        switch (commandUpper) {
+        case "ADD":
+            try {
+                cmd = parseAdd(commandAndRemaining[1].strip(), taskList);
+                break;
+            } catch (IndexOutOfBoundsException e) {
+                throw new InvalidCommandException(AddCommand.getUsage());
+            }
+        case "LIST":
+            cmd = new ListCommand(taskList);
+            break;
+        case "MARK":
+            try {
+                cmd = parseMark(commandAndRemaining[1].strip(), taskList);
+                break;
+            } catch (IndexOutOfBoundsException | NumberFormatException e) {
+                throw new InvalidCommandException(MarkCommand.getUsage());
+            }
+        case "UNMARK":
+            try {
+                cmd = parseUnmark(commandAndRemaining[1].strip(), taskList);
+                break;
+            } catch (IndexOutOfBoundsException | NumberFormatException e) {
+                throw new InvalidCommandException(UnmarkCommand.getUsage());
+            }
+        case "DELETE":
+            try {
+                cmd = parseDelete(commandAndRemaining[1].strip(), taskList);
+                break;
+            } catch (IndexOutOfBoundsException | NumberFormatException e) {
+                throw new InvalidCommandException(DeleteCommand.getUsage());
+            }
+        case "END":
+            cmd = new ExitCommand();
+            break;
+        case "FIND":
+            try {
+                cmd = new FindCommand(taskList, commandAndRemaining[1]);
+            } catch (IndexOutOfBoundsException e) {
+                throw new InvalidCommandException(FindCommand.getUsage());
+            }
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
```

Comment:
I am very glad you have ensured the non-indentation of switch-case statements which is not the case for many since it is a bit strange to us. This is a clear sign that you have done your due diligence in checkstyling your code which is reflected everywhere else too! Good job!